### ISSUE TITLE
joins and default values

### ISSUE THREAD

**Issue by LolusKekus** [coleifer/peewee#1367]
Why peewee returns default values for joined tabels when i forgot to select this field?

I have 2 models:
```
class ModelOne(Model):
    fieldOne = IntegerField(default=666)
    fieldTwo = IntegerField()

class ModelTwo(Model):
    modelOne = ForeignKeyField(ModelOne)
    someField = IntegerField()
```
For the begininng I want to query ModelTwo with all fields in it and fieldTwo from ModelOne:

```
qs = ModelTwo.select(ModelTwo, ModelOne.fieldTwo).join(ModelOne)
for i in qs:
    print(i.someField, i.modelone.fieldTwo)
```
Ok, it's work, i'm happy. BUT suddenly i understand that i need fieldOne from ModelOne too.
I don't know, maybe becaus of my inattentive, i do next thing:

```
qs = ModelTwo.select(ModelTwo, ModelOne.fieldTwo).join(ModelOne)
for i in qs:
    print(i.someField, i.modelone.fieldTwo, i.modelone.fieldOne)
```
I'm completely forgot to insert **ModelOne.fieldOne** in **select()** section and what i did get?

**default** value for **fieldOne**: 666

I think this isn't right. Maybe better raise an error that i forgot to add fieldOne in select section?

**Comment by coleifer** [comment]
I understand your thinking, but unfortunately I'm worried about the implications of modifying this behavior in the 2.x branch. Perhaps I can introduce such a change in 3.x, though?

**Comment by coleifer** [comment]
Fixed in 3.0a. No exception is raised, `None` is simply returned as the field value.
